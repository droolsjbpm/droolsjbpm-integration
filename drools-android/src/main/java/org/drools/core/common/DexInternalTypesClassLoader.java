package org.drools.core.common;

import org.drools.android.MultiDexClassLoader;

/**
 * @author kedzie
 */
public class DexInternalTypesClassLoader extends MultiDexClassLoader {

    private final ProjectClassLoader projectClassLoader;

    public DexInternalTypesClassLoader(ProjectClassLoader projectClassLoader) {
        super(null);
        this.projectClassLoader = projectClassLoader;
    }

    public Class<?> defineClass(String name, byte[] bytecode) {
        int lastDot = name.lastIndexOf( '.' );
        if (lastDot > 0) {
            String pkgName = name.substring( 0, lastDot );
            if (getPackage( pkgName ) == null) {
                definePackage( pkgName, "", "", "", "", "", "", null );
            }
        }
        return super.defineClass(name, bytecode);
    }

    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
        try {
            return loadType(name, resolve);
        } catch (ClassNotFoundException cnfe) {
            synchronized(projectClassLoader) {
                try {
                    return projectClassLoader.internalLoadClass(name, resolve);
                } catch (ClassNotFoundException cnfe2) {
                    return projectClassLoader.tryDefineType(name, cnfe);
                }
            }
        }
    }

    private Class<?> loadType(String name, boolean resolve) throws ClassNotFoundException {
        return super.loadClass(name, resolve);
    }
}
