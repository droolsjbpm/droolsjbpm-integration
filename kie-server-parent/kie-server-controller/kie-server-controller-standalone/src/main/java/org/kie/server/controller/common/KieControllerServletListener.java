/*
 * Copyright 2017 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
package org.kie.server.controller.common;

import java.util.ServiceLoader;

import javax.servlet.ServletContextEvent;
import javax.servlet.ServletContextListener;
import javax.servlet.annotation.WebListener;

import org.kie.server.controller.api.service.PersistingServerTemplateStorageService;
import org.kie.server.controller.api.storage.KieServerTemplateStorage;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@WebListener
public class KieControllerServletListener implements ServletContextListener {

    private static final Logger logger = LoggerFactory.getLogger(KieControllerServletListener.class);
    private KieServerTemplateStorage storage;
    
    @Override
    public void contextDestroyed(ServletContextEvent event) {
        if (storage != null) {
            storage.close();
            
            logger.debug("Template storage {} closed successfully", storage);
        }
    }

    @Override
    public void contextInitialized(ServletContextEvent event) {
        ServiceLoader<PersistingServerTemplateStorageService> storageServices = ServiceLoader.load(PersistingServerTemplateStorageService.class);
        
        if (storageServices != null && storageServices.iterator().hasNext()) {
            PersistingServerTemplateStorageService storageService = storageServices.iterator().next();
            storage = storageService.getTemplateStorage(); 
            
            logger.debug("Template storage {} initialized successfully", storage);
        }
    }

}
