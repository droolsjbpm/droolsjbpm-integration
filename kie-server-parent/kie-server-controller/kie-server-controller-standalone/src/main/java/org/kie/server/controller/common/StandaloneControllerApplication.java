/*
 * Copyright 2017 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/
package org.kie.server.controller.common;

import java.util.Collections;
import java.util.Set;
import java.util.concurrent.CopyOnWriteArraySet;

import javax.ws.rs.ApplicationPath;
import javax.ws.rs.core.Application;

import org.kie.server.controller.service.StandaloneKieServerControllerImpl;
import org.kie.server.controller.service.StandaloneRuntimeManagementServiceImpl;
import org.kie.server.controller.service.StandaloneSpecManagementServiceImpl;


@ApplicationPath("/")
public class StandaloneControllerApplication extends Application {
    
    private final Set<Object> instances;
    
    public StandaloneControllerApplication() {
        instances = new CopyOnWriteArraySet<Object>() {
            private static final long serialVersionUID = 1763183096852523317L;
            {                
                add(new StandaloneKieServerControllerImpl());
                add(new StandaloneSpecManagementServiceImpl());
                add(new StandaloneRuntimeManagementServiceImpl());
            }
        };
    }

    @Override
    public Set<Class<?>> getClasses() {
        return Collections.emptySet();
    }

    @Override
    public Set<Object> getSingletons() {
        return instances;
    }
}
