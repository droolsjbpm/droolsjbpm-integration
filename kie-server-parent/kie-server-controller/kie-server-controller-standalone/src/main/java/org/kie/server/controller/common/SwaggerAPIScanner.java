/*
 * Copyright 2018 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.kie.server.controller.common;

import javax.servlet.ServletContextEvent;
import javax.servlet.ServletContextListener;
import javax.servlet.annotation.WebListener;

import io.swagger.jaxrs.config.BeanConfig;
import io.swagger.jaxrs.config.DefaultJaxrsScanner;
import io.swagger.jaxrs.config.JaxrsScanner;
import io.swagger.jaxrs.config.SwaggerContextService;
import io.swagger.jaxrs.config.SwaggerScannerLocator;
import org.kie.server.controller.api.KieServerControllerConstants;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@WebListener
public class SwaggerAPIScanner implements ServletContextListener {

    public static final Boolean SWAGGER_DISABLED = Boolean.parseBoolean(System.getProperty(KieServerControllerConstants.KIE_CONTROLLER_SWAGGER_DISABLED,
                                                                                           "false"));

    private static final Logger LOGGER = LoggerFactory.getLogger(SwaggerAPIScanner.class);

    @Override
    public void contextInitialized(ServletContextEvent sce) {
        if (SWAGGER_DISABLED) {
            return;
        }

        LOGGER.info("Starting Swagger API discovery");

        JaxrsScanner jaxrsScanner = new DefaultJaxrsScanner();
        jaxrsScanner.setPrettyPrint(true);
        /*
         * Set our JAX-RS Scanner with SCANNER_ID_DEFAULT.
         * We need to do this before creating the BeanConfig, as this prevents the BeanConfig to register itself as the default scanner.
         * The first one wins.
         */
        SwaggerScannerLocator.getInstance().putScanner((SwaggerContextService.SCANNER_ID_DEFAULT), jaxrsScanner);

        BeanConfig beanConfig = new BeanConfig();
        beanConfig.setBasePath(sce.getServletContext().getContextPath() + "/rest");
        beanConfig.setVersion("7.0");
        beanConfig.setTitle("Controller API");
        beanConfig.setPrettyPrint(true);
        beanConfig.setScan();
    }

    @Override
    public void contextDestroyed(ServletContextEvent sce) {
    }
}
