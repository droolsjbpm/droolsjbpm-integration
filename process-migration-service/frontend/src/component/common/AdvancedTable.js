import React from "react";

import { orderBy } from "lodash";
import * as sort from "sortabular";
import * as resolve from "table-resolver";
import { compose } from "recompose";

import PropTypes from "prop-types";

import {
  customHeaderFormattersDefinition,
  Table,
  TABLE_SORT_DIRECTION,
  Grid,
  PaginationRow,
  paginate,
  PAGINATION_VIEW,
  defaultSortingOrder
} from "patternfly-react";

export default class AdvancedTable extends React.Component {
  constructor(props) {
    super(props);

    const getSortingColumns = () => this.state.sortingColumns || {};

    const sortableTransform = sort.sort({
      getSortingColumns,
      onSort: selectedColumn => {
        this.setState({
          sortingColumns: sort.byColumn({
            sortingColumns: this.state.sortingColumns,
            sortingOrder: defaultSortingOrder,
            selectedColumn
          })
        });
      },
      // Use property or index dependening on the sortingColumns structure specified
      strategy: sort.strategies.byProperty
    });

    const sortingFormatter = sort.header({
      sortableTransform,
      getSortingColumns,
      strategy: sort.strategies.byProperty
    });

    this.columns = this.props.columns;

    //if the table is sortable, add sorting function, and transforming function to column defination.
    //"Actions" column should not be sortable.
    if (this.props.isSortable) {
      this.columns.map(column => {
        if (column.header.label.toLowerCase().includes("action")) {
          return column;
        } else {
          column.header.transforms.push(sortableTransform);
          column.header.formatters.push(sortingFormatter);
          return column;
        }
      });
    }

    // enables our custom header formatters extensions to reactabular
    this.customHeaderFormatters = customHeaderFormattersDefinition;
    this.state = {
      // Sort the first column in an ascending way by default.
      sortingColumns: {
        name: {
          direction: TABLE_SORT_DIRECTION.ASC,
          position: 0
        }
      },

      // rows
      rows: this.props.rows,

      // pagination default states
      pagination: {
        page: 1,
        perPage: 6,
        perPageOptions: [6, 10, 15]
      },

      // page input value
      pageChangeValue: 1
    };
  }

  onFirstPage = () => {
    this.setPage(1);
  };
  onLastPage = () => {
    const { page } = this.state.pagination;
    const totalPages = this.totalPages();
    if (page < totalPages) {
      this.setPage(totalPages);
    }
  };
  onNextPage = () => {
    const { page } = this.state.pagination;
    if (page < this.totalPages()) {
      this.setPage(this.state.pagination.page + 1);
    }
  };
  onPageInput = e => {
    this.setState({ pageChangeValue: e.target.value });
  };
  onPerPageSelect = eventKey => {
    const newPaginationState = Object.assign({}, this.state.pagination);
    newPaginationState.perPage = eventKey;
    newPaginationState.page = 1;
    this.setState({ pagination: newPaginationState });
  };
  onPreviousPage = () => {
    if (this.state.pagination.page > 1) {
      this.setPage(this.state.pagination.page - 1);
    }
  };

  onSubmit = () => {
    this.setPage(this.state.pageChangeValue);
  };
  setPage = page => {
    if (!isNaN(page) && page !== "" && page > 0 && page <= this.totalPages()) {
      const newPaginationState = Object.assign({}, this.state.pagination);
      newPaginationState.page = page;
      this.setState({ pagination: newPaginationState, pageChangeValue: page });
    }
  };
  currentRows() {
    const { rows, sortingColumns, pagination } = this.state;
    const { columns } = this;
    return compose(
      paginate(pagination),
      sort.sorter({
        columns,
        sortingColumns,
        sort: orderBy,
        strategy: sort.strategies.byProperty
      })
    )(rows);
  }
  totalPages = () => {
    const { perPage } = this.state.pagination;
    return Math.ceil(this.props.rows.length / perPage);
  };
  render() {
    const { pagination, sortingColumns, pageChangeValue } = this.state;
    const { columns } = this;
    const sortedPaginatedRows = this.currentRows();

    return (
      <Grid fluid>
        <Table.PfProvider
          striped
          bordered
          hover
          dataTable
          columns={columns}
          components={{
            header: {
              cell: cellProps =>
                this.customHeaderFormatters({
                  cellProps,
                  columns,
                  sortingColumns,
                  rows: sortedPaginatedRows.rows,
                  onSelectAllRows: this.onSelectAllRows
                })
            }
          }}
        >
          <Table.Header headerRows={resolve.headerRows({ columns })} />
          <Table.Body rows={sortedPaginatedRows.rows} rowKey="id" />
        </Table.PfProvider>
        <PaginationRow
          viewType={PAGINATION_VIEW.TABLE}
          pagination={pagination}
          pageInputValue={pageChangeValue}
          amountOfPages={sortedPaginatedRows.amountOfPages}
          itemCount={sortedPaginatedRows.itemCount}
          itemsStart={sortedPaginatedRows.itemsStart}
          itemsEnd={sortedPaginatedRows.itemsEnd}
          onPerPageSelect={this.onPerPageSelect}
          onFirstPage={this.onFirstPage}
          onPreviousPage={this.onPreviousPage}
          onPageInput={this.onPageInput}
          onNextPage={this.onNextPage}
          onLastPage={this.onLastPage}
          onSubmit={this.onSubmit}
          className="migration-log-table-pagination"
        />
      </Grid>
    );
  }
}

AdvancedTable.propTypes = {
  rows: PropTypes.array.isRequired,
  columns: PropTypes.array.isRequired,
  isSortable: PropTypes.bool.isRequired
};
