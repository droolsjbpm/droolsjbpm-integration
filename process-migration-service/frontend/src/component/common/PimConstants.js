export const USE_MOCK_DATA = false;
export const BACKEND_URL = "/rest";
