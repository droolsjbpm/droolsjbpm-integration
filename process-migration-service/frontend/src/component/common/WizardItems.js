export const AddPlanItems = [
  {
    step: 1,
    label: "1",
    title: "Define Plan",
    subSteps: [
      {
        subStep: "1.1",
        label: "1A.",
        title: "Plan Details",
        contents: {}
      }
    ]
  },
  {
    step: "2",
    label: "2",
    title: "Process Definition",
    subSteps: [
      {
        subStep: "2.1",
        label: "2A.",
        title: "Process Definition",
        contents: {}
      }
    ]
  },
  {
    step: "3",
    label: "3",
    title: "Node Mapping",
    subSteps: [
      {
        subStep: "3.1",
        label: "3A.",
        title: "Mapping",
        contents: {}
      }
    ]
  },
  {
    step: 4,
    label: "4",
    title: "Review & Submit",
    subSteps: [
      {
        subStep: "4.1",
        label: "4A.",
        title: "Review"
      }
    ]
  },
  {
    step: 5,
    label: "5",
    title: "Check Response",
    subSteps: [
      {
        subStep: "5.1",
        label: "5A.",
        title: "Response"
      }
    ]
  }
];

export const ExecuteMigrationItems = [
  {
    step: 1,
    label: "1",
    title: "Select Running Instances",
    subSteps: [
      {
        subStep: "1.1",
        label: "1A.",
        title: "Plan Details",
        contents: {}
      }
    ]
  },
  {
    step: "2",
    label: "2",
    title: "Choose Execution Time",
    subSteps: [
      {
        subStep: "2.1",
        label: "2A.",
        title: "Process Definition",
        contents: {}
      }
    ]
  },
  {
    step: 3,
    label: "3",
    title: "Review & Submit",
    subSteps: [
      {
        subStep: "3.1",
        label: "3A.",
        title: "Review"
      }
    ]
  },
  {
    step: 4,
    label: "4",
    title: "Check Response",
    subSteps: [
      {
        subStep: "4.1",
        label: "4A.",
        title: "Response"
      }
    ]
  }
];
