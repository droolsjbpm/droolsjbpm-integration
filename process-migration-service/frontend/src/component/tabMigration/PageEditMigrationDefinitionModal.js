import React from "react";
import axios from "axios";

import { Button } from "patternfly-react";
import { Modal } from "patternfly-react";
import { Icon } from "patternfly-react";
import { OverlayTrigger } from "patternfly-react";
import { Tooltip } from "patternfly-react";

import { BACKEND_URL } from "../common/PimConstants";
import PageMigrationScheduler from "../tabMigrationPlan/wizardExecuteMigration/PageMigrationScheduler";

export default class PageEditMigrationDefinitionModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: this.props.rowData.id,
      planId: this.props.rowData.definition.planId,
      processInstanceIds: this.props.rowData.definition.processInstanceIds,
      kieserverId: this.props.rowData.definition.kieserverId,
      scheduledStartTime: this.props.rowData.definition.execution
        .scheduledStartTime,
      callbackUrl: this.props.rowData.definition.execution.callbackUrl,
      showEditDialog: false
    };
  }

  convertFormDataToJson() {
    const execution = {
      type: "ASYNC"
    };

    if (
      this.state.scheduledStartTime !== null &&
      this.state.scheduledStartTime !== ""
    ) {
      execution.scheduledStartTime = this.state.scheduledStartTime;
    }
    if (this.state.callbackUrl !== null && this.state.callbackUrl !== "") {
      execution.callbackUrl = this.state.callbackUrl;
    }

    const formData = {
      planId: this.state.planId,
      kieserverId: this.state.kieserverId,
      processInstanceIds: "[" + this.state.processInstanceIds + "]",
      execution: execution
    };

    var jsonStr = JSON.stringify(formData, null, 2);

    if (jsonStr !== null && jsonStr !== "") {
      //replace "[ to [
      jsonStr = jsonStr.replace('"[', "[");

      //replace ]" to ]
      jsonStr = jsonStr.replace(']"', "]");
    }

    return jsonStr;
  }

  setCallbackUrl = url => {
    this.setState({
      callbackUrl: url
    });
  };

  setScheduleStartTime = startTime => {
    this.setState({
      scheduledStartTime: startTime
    });
  };

  hideEditDialog = () => {
    this.setState({ showEditDialog: false });
  };

  openEditMigration = () => {
    this.setState({
      showEditDialog: true
    });
  };

  submit = () => {
    const serviceUrl = BACKEND_URL + "/migrations/" + this.state.id;
    const migrationDefinitionJsonStr = this.convertFormDataToJson();
    axios
      .put(serviceUrl, migrationDefinitionJsonStr, {
        headers: {
          "Content-Type": "application/json"
        }
      })
      .then(() => {
        this.hideEditDialog();
      });
  };

  render() {
    const tooltipEdit = (
      <Tooltip id="tooltip">
        <div>Edit</div>
      </Tooltip>
    );

    const defaultBody = (
      <div>
        <div className="form-horizontal">
          <div className="form-group">
            <label className="col-md-4 control-label"> Plan ID</label>
            <div className="col-md-8">{this.state.planId}</div>
          </div>
          <div className="form-group">
            <label className="col-md-4 control-label">
              {" "}
              Process Instances ID
            </label>
            <div className="col-md-8">
              {JSON.stringify(this.state.processInstanceIds)}
            </div>
          </div>
          <div className="form-group">
            <label className="col-md-4 control-label"> KIE Server ID</label>
            <div className="col-md-8">{this.state.kieserverId}</div>
          </div>
        </div>

        <PageMigrationScheduler
          setCallbackUrl={this.setCallbackUrl}
          setScheduleStartTime={this.setScheduleStartTime}
        />
      </div>
    );

    return (
      <div>
        <OverlayTrigger overlay={tooltipEdit} placement={"bottom"}>
          <Button bsStyle="link" onClick={this.openEditMigration}>
            <Icon type="fa" name="edit" />
          </Button>
        </OverlayTrigger>
        <Modal
          show={this.state.showEditDialog}
          onHide={this.hideEditDialog}
          size="lg"
        >
          <Modal.Header>
            <Modal.CloseButton onClick={this.hideEditDialog} />
            <Modal.Title>Edit Migration Definition</Modal.Title>
          </Modal.Header>
          <Modal.Body>{defaultBody}</Modal.Body>
          <Modal.Footer>
            <Button
              bsStyle="default"
              className="btn-cancel"
              onClick={this.hideEditDialog}
            >
              Cancel
            </Button>
            <Button bsStyle="primary" onClick={this.submit}>
              Submit
            </Button>
          </Modal.Footer>
        </Modal>
      </div>
    );
  }
}
