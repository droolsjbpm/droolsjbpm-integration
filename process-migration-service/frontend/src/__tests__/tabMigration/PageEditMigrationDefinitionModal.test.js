import renderer from "react-test-renderer";
import React from "react";
import { MockupData_Migrations_Definitions } from "../../component/common/MockupData";
import PageEditMigrationDefinitionModal from "../../component/tabMigration/PageEditMigrationDefinitionModal";

test("PageEditMigrationDefinitionModal renders correctly using snapshot", () => {
  const mockData = MockupData_Migrations_Definitions;
  const tree = renderer
    .create(<PageEditMigrationDefinitionModal rowData={mockData[0]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
