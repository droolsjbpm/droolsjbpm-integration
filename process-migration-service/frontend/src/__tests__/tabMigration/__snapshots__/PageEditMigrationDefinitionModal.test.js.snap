// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageEditMigrationDefinitionModal renders correctly using snapshot 1`] = `
<div>
  <button
    className="btn btn-link"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <span
      aria-hidden={true}
      className="fa fa-edit"
    />
  </button>
</div>
`;
